processor/src/decorators/mod.rs

```diff
@@ -1,14 +1,21 @@
-use super::{AdviceInjector, ExecutionError, Felt, Process, StarkField};
+use super::{AdviceInjector, Decorator, ExecutionError, Felt, Process, StarkField};
 
 // DECORATORS
 // ================================================================================================
 
 impl Process {
+    /// Executes the specified decorator
+    pub(super) fn execute_decorator(&mut self, decorator: Decorator) -> Result<(), ExecutionError> {
```

Comment:
Should this take a reference to the decorator? I'd imagine decorators being pretty heaving in the future - so, if we can pass them by reference, it may be more performant.

Reply:
Thanks. I missed this. Fixed.